### DISCUSSION TITLE
SSH-Key

### DISCUSSION THREAD

**Discussion by mcmlix** [mKeRix/room-assistant#793] [Q&A]
I have been running RA for about a year. Great project kudos to you sir.  I have three pi zeros that I have been individually updating.
I now would like to use ansible, but all my pi's have a ssh-key.
I am really confused at how and were to put the key user and location.
I use 
user : pi
and a key file in ./ssh
I'm using Arch Linux..Non of that Windows crap.
Could you  give me some help. Please.
Thank you in advance.

**Comment by mKeRix** [comment]
On Linux and Mac Ansible should simply pick up the SSH key from `~/.ssh/id_rsa`, like any other SSH client would. That's how it works for me at least. Then you only need to provide the username, which you can e.g. do via the command line (`-u pi`):

```shell
ansible-playbook -i hosts.yml -u pi room-assistant.ym
```

Ansible also offers some [documentation](https://docs.ansible.com/ansible/latest/user_guide/connection_details.html) on this topic.

**Comment by mcmlix** [comment]
hi @mKeRix thank you for your answer. 
i will try ansible-playbook -i hosts.yml -u pi room-assistant.yml --check -e ansible_ssh_port=2332
I'm getting the following errors.
[WARNING]: Unable to parse /home/XXXX/ansible-playbooks/hosts.yml as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
I've really never used Ansible. I think I have to get more familiar with it. I probably have the port portion wrong.
I just have the host.yaml file.
